client/orderbook/orderbook.go

```diff
@@ -627,16 +627,25 @@ func (ob *OrderBook) SetMatchesSummary(matches map[uint64]uint64, ts uint64) {
 		}
 		i++
 	}
+
+	ob.matchSummaryMtx.Lock()
 	if ob.matchesSummary == nil {
 		ob.matchesSummary = newMatchesSummary
 		return
 	}
 	ob.matchesSummary = append(newMatchesSummary, ob.matchesSummary...)
+	// if len is greater than 100, slice the first 100.
+	if len(ob.matchesSummary) > 100 {
+		ob.matchesSummary = ob.matchesSummary[0:100]
+	}
+	ob.matchSummaryMtx.Unlock()
 }
```

Comment:
This will only save tho older ones correct? I think you want to throw away the older ones, so `ob.matchesSummary = ob.matchesSummary[100:]`

https://go.dev/play/p/_Dg-K0wEnNA

Reply:
also, this would make more sense if len is over 1000 or so, otherwise would delete almost everything whenever it hits 100.

Reply:
Yeah 100 is too small. 

Now I am doing 1000 and if it is greater than 1100.

Thanks for the suggestion